### ISSUE TITLE
Internal module name collision with Python built-in

### ISSUE THREAD

**Issue by astralcai** [microsoft/pyright#10991]
**Describe the bug**

Pyright incorrectly resolves a local, internal module named `math` to the built-in Python `math` module, leading to false-positive type errors ("... is not a known attribute of module math") when accessing attributes defined in the internal module.

**Code or Screenshots**

With the following import statement
```python
from pennylane import math
```
at the top of the file, when trying to access functions inside the internal `math` module,

<img width="577" height="128" alt="Image" src="https://github.com/user-attachments/assets/f65d366d-a17a-4f1b-ad61-8711c1088b25" />

The same error is raised in neovim using pyright as the LSP

<img width="901" height="79" alt="Image" src="https://github.com/user-attachments/assets/df154cf1-c7fc-434a-a868-622b37d73c1d" />

**Pyright Version**

Version 1.1.40

**Comment by erictraut** [comment]
Can you provide more details about how to repro this? In particular, describe the layout of the relevant files locally in your project and the minimum relevant statements in each file.

If you right-click on `math` in the import statement and choose "Go To Declaration", where does it take you? 

You mentioned that you suspect a collision with the stdlib `math` module. What leads you to think that's the case? Are you also importing the stdlib `math` module in the same file?

**Comment by astralcai** [comment]
I'm trying to construct a minimal non-working example and find that I'm unable, but in the repository that I'm working in (https://github.com/PennyLaneAI/pennylane), if you navigate to a line where `math` is imported, say [here](https://github.com/PennyLaneAI/pennylane/blob/fd44655ea3e3f462519d0228c4873049d6cef065/pennylane/workflow/qnode.py#L30), and click "Go To Declaration", it takes me to `math.pyi` in `stdlib`, which is why I suspected that this was the issue.

**Comment by astralcai** [comment]
I figured it out, there was a bad `from ... import *` which contained a local import of the builtin `math` module, which overrides the first import of `from pennylane import math` in the `__init__.py`. I'm closing the issue, sorry for the trouble.
